youtube_dl/YoutubeDL.py

```diff
@@ -1291,6 +1289,22 @@ def process_info(self, info_dict):
             self.report_error('unable to create directory ' + compat_str(err))
             return
 
+        filesystem_enc = sys.getfilesystemencoding()
+        if filesystem_enc is None:
+            filesystem_enc = 'utf-8'
+        max_filename_len = 200
+        if sys.platform.startswith('linux'):
```

Comment:
>os.pathconf and os.pathconf_names both available on unix

Not necessarily, [they can be disabled](https://github.com/python/cpython/blob/9586a26986ab6fe8baac15d6db29b5e19c09ba65/Modules/posixmodule.c#L10483-L10489):
```
>>> import os
>>> dir(os.pathconf)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module object has no attribute 'pathconf'
```